Update pre-commit hooks and add ruff

sethrj opened the pull request:
This adds a python formatter (ruff) for the python scripts we have scattered around, and it bumps slightly the clang-format version. I added but disabled the ruff linter: it caught an unused code error, but it has some false positives (e.g., sphinx conf.py) that I don't want to deal with right now.

sethrj committed:
Update clang format pre-commit version

sethrj committed:
Add ruff for python file checking

sethrj committed:
Format python files

sethrj committed:
Remove disused case for ORANGE JSON

sethrj committed:
Remove ruff check due to minor sphinx/orange/monkey compatibility

sethrj committed:
Reorder hooks

sethrj committed:
Add ignore-revs from previous pre-commit application

sethrj committed:
Revert "Format python files"

This reverts commit 0634d946c8d504306e87c2ca91639cc825fcca2a.

sethrj committed:
Update pre-commit message

sethrj committed:
Add ignore-rev for ruff format

sethrj committed:
Merge remote-tracking branch 'upstream/develop' into pre-commit-update

pcanal commented on .pre-commit-config.yaml:
```diff
@@ -7,30 +7,41 @@ repos:
 
 ci:
+  autofix_commit_msg: |
+    Apply pre-commit hooks: please install pre-commit locally!
```
If I understand correctly, the default is: 
```
'[pre-commit.ci] auto fixes [...]'
```
The new message may (or may not) have lost the 'information' that some fixes were needed and have been automatically applied.

pcanal replied:
Before
<img width="925" height="105" alt="Screenshot 2025-10-22 at 19 34 46" src="https://github.com/user-attachments/assets/cc6baba0-092e-4fd2-b9eb-90890f1dc2f6" />
and after
<img width="880" height="96" alt="Screenshot 2025-10-22 at 19 39 08" src="https://github.com/user-attachments/assets/79243eb1-ff25-44bb-9826-e31824ce0b68" />

sethrj replied:
I think the second one does provide more information? The fact that's pre-commit.ci is obvious from the icon and the link. The commit message is what it does in the imperative-style commit message that is supposed to be git standard ("Apply pre-commit hooks") and then it goes on to tell you what to do (please install pre-commit locally). Do you find the second message more confusing? Is there a way to improve on it?

pcanal replied:
I don't have a strong opinion and you can leave as is.  Here is a suggestion:
```suggestion
    Apply pre-commit hooks' fixes. (Please install pre-commit locally to avoid this commit!)
```

pcanal reviewed (approved):
LGTM.

sethrj committed:
Update autofix message

sethrj committed:
Introduce whitespace error to be fixed

sethrj committed:
Introduce whitespace error to be fixed

sethrj committed:
Merge remote-tracking branch 'upstream/develop' into pre-commit-update

pre-commit-ci[bot] committed:
Apply fixes from pre-commit hooks

This commit means that you must (please!) install pre-commit
on your development machine and run `pre-commit install --install-hooks`.
For more information, see
https://celeritas-project.github.io/celeritas/user/development/style.html#formatting

Autogenerated: https://pre-commit.ci

sethrj committed:
Try to make commit message more obvious

pre-commit-ci[bot] committed:
Apply fixes from pre-commit hooks: see detailed commit message →

This commit means that you must (please!) install pre-commit
on your development machine and run `pre-commit install --install-hooks`.
For more information, see
https://celeritas-project.github.io/celeritas/user/development/style.html#formatting

Autogenerated: https://pre-commit.ci

sethrj commented:
Thanks for the feedback @pcanal , I think the commit message and associated instructions are quite clear now 😄
